src/App.js

```diff
@@ -6,14 +6,32 @@ import { Routes, Route } from "react-router-dom";
 import FavoritesPage from './components/FavoritesPage';
 import RandomizerPage from './components/RandomizerPage';
 import ExtrasPage from './components/ExtrasPage';
+import { useEffect, useState } from "react";
 
 function App() {
 
+  const apiURL="https://rickandmortyapi.com/api/character";
+  const [characters, setCharacters] = useState([]);
+  const [showMore, setShowMore] = useState(false);
+
+  const fetchCharacters =() =>{
+    fetch(apiURL).then((response) => response.json()).then((data) => setCharacters(data.results))
+  }
+
+  useEffect(() => {
+    fetchCharacters();
+  }, []);
+
+  const toggleInfo = () => {
+    setShowMore(!showMore);
+  }
+
+
   return (
     <>
       <Header />
       <Routes>
-        <Route path="/" element={<CharacterPage />}/>
+        <Route path="/" element={<CharacterPage characters={characters} toggleInfo={toggleInfo}/>}/>
```

Comment:
YOu are not giving this component the showMore prop, but the component expects it